sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -160,6 +160,13 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       UpdateNullabilityInAttributeReferences)
   }
 
+  def nonExcludableBatches: Seq[String] =
+    "Eliminate Distinct" ::
+      "Finish Analysis" ::
+      "Replace Operators" ::
+      "Pullup Correlated Expressions" ::
+      "RewriteSubquery" :: Nil
```

Comment:
We use not rule names but batch names in this black list?

Reply:
I'll change to rule black list.